[BUG]: AnkiDroid 2.17alpha6 AnkiDroid Crashes when accessed from background processes

### Checked for duplicates?

- [X] This issue is not a duplicate

### What are the steps to reproduce this bug?

There are multiple apps that can export Anki cards to AnkiDroid.
- Jidoujisho (https://github.com/lrorpilla/jidoujisho)
- Yomichan w/ Kiwi browser + AnkiConnect Android ( https://learnjapanese.moe/yomichan/ + https://github.com/KamWithK/AnkiconnectAndroid)

Doing any Anki related features shows that AnkiDroid has stopped working

### Expected behaviour

Same as the working 2.16.5, it should not crash and allow for cards to be added.		

### Actual behaviour

"AnkiDroid has stopped working" popup occurs

### Debug info

```text
AnkiDroid Version = 2.17alpha6 (393f61b229a26ec97eb13c83f0d1f09652f776c2)

Android Version = 9

ProductFlavor = full

Manufacturer = samsung

Model = SM-N950U1

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 9; SM-N950U1 Build/PPR1.180610.011; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/119.0.6045.163 Mobile Safari/537.36

ACRA UUID = 85c60929-c55d-4046-ae16-545886ccf69b

Crash Reports Enabled = true

DatabaseV2 Enabled = true
```


### (Optional) Anything else you want to share?

_No response_

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

Hi, could you ensure that crash reports are enabled? I can only see two reports from 2.16.5


```
Caused by: org.json.JSONException: No value for new
	at org.json.JSONObject.get(JSONObject.java:392)
	at org.json.JSONObject.getJSONObject(JSONObject.java:612)
	at com.ichi2.libanki.sched.SchedV2._newConf(SourceFile:18)
```

> Hi, could you ensure that crash reports are enabled? I can only see two reports from 2.16.5
> 
> ```
> Caused by: org.json.JSONException: No value for new
> 	at org.json.JSONObject.get(JSONObject.java:392)
> 	at org.json.JSONObject.getJSONObject(JSONObject.java:612)
> 	at com.ichi2.libanki.sched.SchedV2._newConf(SourceFile:18)
> ```

Sorry, It's not crashing on 2.16.5 like i mentioned in the original post, but when it crashes in the background on the alpha versions theres no crash report popup unfortunately

--- a/AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidApp.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidApp.kt
@@ -48,6 +48,7 @@ import com.ichi2.anki.preferences.sharedPrefs
 import com.ichi2.anki.services.BootService
 import com.ichi2.anki.services.NotificationService
 import com.ichi2.anki.ui.dialogs.ActivityAgnosticDialogs
+import com.ichi2.annotations.NeedsTest
 import com.ichi2.compat.CompatHelper
 import com.ichi2.utils.*
 import kotlinx.coroutines.CoroutineScope
@@ -156,12 +157,7 @@ open class AnkiDroidApp : Application() {
         )
         CompatHelper.compat.setupNotificationChannel(applicationContext)
 
-        if (Build.FINGERPRINT != "robolectric") {
-            // Prevent sqlite throwing error 6410 due to the lack of /tmp on Android
-            Os.setenv("TMPDIR", cacheDir.path, false)
-            // Load backend library
-            System.loadLibrary("rsdroid")
-        }
+        makeBackendUsable(this)
 
         // Configure WebView to allow file scheme pages to access cookies.
         if (!acceptFileSchemeCookies()) {
@@ -412,6 +408,23 @@ open class AnkiDroidApp : Application() {
             with(field) {
                 isAccessible = true
                 set(field, value)
+            }
+        }
+
+        /** Load the libraries to allow access to Anki-Android-Backend */
+        @NeedsTest("Not calling this in the ContentProvider should have failed a test")
+        fun makeBackendUsable(context: Context) {
+            // Robolectric uses RustBackendLoader.ensureSetup()
+            if (Build.FINGERPRINT == "robolectric") return
+
+            try {
+                // Prevent sqlite throwing error 6410 due to the lack of /tmp on Android
+                Os.setenv("TMPDIR", context.cacheDir.path, false)
+                // Load backend library
+                System.loadLibrary("rsdroid")
+            } catch (e: Error) {
+                // this routine /may/ be called more than once: we'll know if/when it occurs
+                Timber.e("Failed to open usable backend", e)
             }
         }
 

--- a/AnkiDroid/src/main/java/com/ichi2/anki/provider/CardContentProvider.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/provider/CardContentProvider.kt
@@ -150,6 +150,7 @@ class CardContentProvider : ContentProvider() {
     override fun onCreate(): Boolean {
         // Initialize content provider on startup.
         Timber.d("CardContentProvider: onCreate")
+        AnkiDroidApp.makeBackendUsable(context!!)
         return true
     }
 

